### ISSUE TITLE
Extract text from a pdf for testing

### ISSUE THREAD

**Issue by Ignaciort91** [danfickle/openhtmltopdf#426]
I need to extract some text to verify if the PDF output is correct. I am using PDFStreamParser of PDFBox to extract the text, but the result is bad due to coding, try a PDF created by me and the PDF showcase with the same result. Can I configure the encoding before compiling the pdf? or set the correct default encoding in PDFStreamParser? Is it possible to extract text that way?

Here is an excerpt from the code:

```
PDDocument pdfDocument = PDDocument.load(new File(url));
PDPageTree allPages = pdfDocument.getDocumentCatalog().getPages();
PDPage page = allPages.get(0);
PDFStreamParser parser = new PDFStreamParser(page);
parser.parse();
List tokens = parser.getTokens();
System.out.println("Start...");
for (int i = 0; i < tokens.size(); i++) {
	Object next = tokens.get(i);
	if (next instanceof COSString) {
		COSString str = (COSString) next;
		String text = str.getString();
		System.out.println("TEXT: " + text);
	}
}
System.out.println("End...");

Output:
Start...
TEXT:  ? ` U ^ 8 D = < d _ @ 4 6 ˚ S _ ]  C X _ g S Q c U
TEXT:  2 S H Q + 7 0 / W R 3 ' )  F R P
...
End...
```

**Comment by danfickle** [comment]
I think you want to use a `PDFTextStripper`, or if it doesn't suit your needs, see its source code.

**Comment by Ignaciort91** [comment]
Finally I solved it in another way, to give some context: the idea was to validate if there is a certain text with a specific alternative text.
The first case was resolved using PDFTextStripper.
The second one is looking for all PDStructureElement validating getAlternateDescription.
While it is not the best solution since both things are validated separately, it meets the validation requirements.

Anyway thanks for your reply.
